config.go

```diff
@@ -49,6 +49,7 @@ func newDefaultConfig() *Config {
 		MaxIdleConns:                100,
 		MaxIdleConnsPerHost:         50,
 		OAuthURI:                    "/oauth",
+		OAuthAuthMethod:             "basic",
```

Comment:
Rename to 'ClientAuthMethod'. Supported values should be 'secret-basic' and 'secret-body'. It should be prefixed with 'secret-' to differentiate between other authentication mechanisms that may be added in the future. Also, it should be 'body' not 'post' (both basic and body are after all sent in a post).